src/main/scala/zio/prelude/AddMultiplyShape.scala

```diff
@@ -0,0 +1,100 @@
+package zio.prelude
+
+import zio.prelude.newtypes.{ Prod, Sum }
+
+trait AddMultiplyShape[A, +Addition[x] <: Associative[x], +Multiplication[x] <: Associative[x]] {
+
```

Comment:
As an experiment, we could see about moving `Addition` and `Multiplication` to type members, and using the `Aux` pattern (refinement typing), introduce type aliases for common combinations.

This is only practical if Scala can find the implicits for the common combinations of properties of the two binary operators.

The possible benefits (if workable):

 - Fewer type parameters
 - Easy to make type aliases for common combinations and require those concisely

Reply:
I actually tried that, but there were some downsides and taken all together, I didn't feel the pros outweigh the cons (mainly verboseness and it's an all new concept, not yet used in ZIO Prelude).
But I didn't do it with the `Aux` helpers!

I will give it another try, now with `Aux` and will report with results :+1:

Reply:
I've reworked the codebase, changing those type parameters to type member while utilizing the `Aux` pattern, where handy.
And I think it's an improvement. The code feels more Scala-y and I like it :sunglasses: 
If you're not worried it would feel overwhelming to new ZIO Prelude users, I think we can go forward with it.

Please have a look yourself and tell me what you think :wink: